build/NuGetAdditionalFiles/RunCsc

```diff
@@ -1,5 +1,15 @@
 #!/bin/sh
 
+set -e
+set -u
+
 THISDIR=$(dirname $0)
 
-dotnet $THISDIR/csc.dll "$@"
+# The expression ${VAR:+TEXT} expands to:
+#  If VAR is null or empty => empty string
+#  If VAR is non-empty => TEXT
+# Do this expression to make sure HOST_PATH ends with a slash,
+#  but if it's unset, have it be completely empty (look up dotnet in PATH)
+HOST_PATH=${DOTNET_HOST_PATH:+${DOTNET_HOST_PATH-}/}
```

Comment:
It *is* doing that. `@{VAR:+TXT}` is equivalent to `if (defined VAR) then (TXT) else ("")` (pseudo-syntax).

The reason why it's not just directly `${DOTNET_HOST_PATH}dotnet` is: what if DOTNET_HOST_PATH doesn't end in a slash? Say, `/some/path`, we'd then be invoking `/some/pathdotnet`.

If we wrote `${DOTNET_HOST_PATH}/dotnet`, then an empty host path would cause `/dotnet` to be invoked, not `dotnet`.

The reason it's not an if statement is because bash doesn't have a "is this variable defined" builtin - the way to check if a var is defined is to use one of these features like `:+`, so if we're doing that anyway, why not just jump straight to the right thing.